src/chat/_chatwidget.py

```diff
@@ -224,6 +224,45 @@ def join(self, channel):
     def log_event(self, e):
         self.serverLogArea.appendPlainText("[%s: %s->%s]" % (e.eventtype(), e.source(), e.target()) + "\n".join(e.arguments()))
 
+    def updateAutojoinIRCMenu(self):
+        self.client.menuAutojoinIRC.clear()
+        
+        autojoin_channels = Settings.get("chat/autojoin_channels", [], type=str).split(".")
+        autojoin_channels.remove('')
+        
+        for channel in self.channels:
+            if not self.channels[channel].private and not channel == "#aeolus":    
+                action = self.client.menuAutojoinIRC.addAction(channel)
+                action.setCheckable(True)
+                # We set the state of the items before assigning an function to them, since
+                # otherwise the function would get called to early
+                if channel in autojoin_channels:
+                    action.setChecked(True)
+                action.toggled.connect(lambda checked, chan=channel: self.updateAutojoinSettings(chan,checked))
+                
+        #now add channels we autojoin, but that we left in the meantime
+        for channel in autojoin_channels:
+            if channel not in self.channels:
+                action = self.client.menuAutojoinIRC.addAction(channel)
+                action.setCheckable(True)
+                action.setChecked(True)
+                action.toggled.connect(lambda checked, chan=channel: self.updateAutojoinSettings(chan,checked))
+                
+        
+
+    def updateAutojoinSettings(self, channel, checked):
+        logger.info("Setting Autojoin Settings for channel %s to %s" %(channel, "True" if checked else "False"))
+        currentSettings = Settings.get("chat/autojoin_channels", [], type=str).split(".")
```

Comment:
Ditto. At this point, since this line is duplicated three times, I suggest you use the `persisted_property` helper I've started using elsewhere in the codebase. Then you can pull out the settings getter into a property, e.g. `ChatWidget.autojoin_channels` which gets 'automatically' persisted when assigned.

If you wrap the persisted property with another property, you could make syntactic sugar to e.g. allow `self.autojoin_channels += ['#my_favourite_room']`